crates/agentgateway/src/mcp/router.rs

```diff
@@ -173,7 +173,9 @@ impl App {
 							Ok(claims) => {
 								// Populate context with verified JWT claims before continuing
 								ctx.with_jwt(&claims);
-								req.headers_mut().remove(http::header::AUTHORIZATION);
+								// NOTE: Authorization header is now kept to allow forwarding to HTTP MCP backends
+								// when backendAuth: passthrough is configured
```

Comment:
The whole point of `backendAuth: passthrough` is that you need to explicitly opt-in to passing through. So we need to **remove it** so if they _don't_ have passthrough, then its not passed through.

If the config isn't working, then its a bug around adding it back; we still need to remove it here.